src/hotspot/cpu/arm/c1_LIRAssembler_arm.cpp

```diff
@@ -152,8 +152,14 @@ void LIR_Assembler::osr_entry() {
   int monitor_offset = (method()->max_locals() + 2 * (number_of_locks - 1)) * BytesPerWord;
   for (int i = 0; i < number_of_locks; i++) {
     int slot_offset = monitor_offset - (i * 2 * BytesPerWord);
-    __ ldr(R1, Address(OSR_buf, slot_offset + 0*BytesPerWord));
-    __ ldr(R2, Address(OSR_buf, slot_offset + 1*BytesPerWord));
+    if (slot_offset >= 4096 - BytesPerWord) {
+      __ add_slow(R2, OSR_buf, slot_offset);
```

Comment:
Can't we check this once before the loop? Or doesn't it just make sense?

Reply:
Hi Sergey. Thanks for looking at this.
This is not performance critical code, and the typical number_of_locks value is 0, so IF inside the FOR loop makes sense here.